python/understack-workflows/understack_workflows/main/sync_nautobot_interfaces.py

```diff
@@ -8,29 +11,51 @@
 
 logger = setup_logger(__name__)
 
-
-def main():
-    parser = arg_parser(__file__)
-    args = parser.parse_args()
-
-    default_nb_url = "http://nautobot-default.nautobot.svc.cluster.local"
-    device_name = args.hostname
-    nb_url = args.nautobot_url or default_nb_url
-    nb_token = args.nautobot_token or credential("nb-token", "token")
-    oob_username = args.oob_username or credential("oob-secrets", "username")
-    oob_password = args.oob_password or credential("oob-secrets", "password")
-
+DEFAULT_NB_URL = "http://nautobot-default.nautobot.svc.cluster.local"
+
+
+def argument_parser(name):
+    parser = argparse.ArgumentParser(
+        prog=os.path.basename(name), description="Nautobot Interface sync"
+    )
+    parser.add_argument(
+        "--device-id", type=UUID, required=True, help="Nautobot device ID"
+    )
+    parser.add_argument("--oob_username", required=False, help="OOB username")
+    parser.add_argument("--oob_password", required=False, help="OOB password")
+    parser.add_argument(
+        "--nautobot_url",
+        required=False,
+        help="Nautobot API %(default)s",
+        default=DEFAULT_NB_URL,
+    )
+    parser.add_argument("--nautobot_token", required=False)
+    return parser
+
+
+def do_sync(device_id: UUID, nb_url: str, nb_token: str, bmc_user: str, bmc_pass: str):
     nautobot = Nautobot(nb_url, nb_token, logger=logger)
-    oob_ip = nautobot.device_oob_ip(device_name)
-    oob = oob_sushy_session(oob_ip, oob_username, oob_password)
+    oob_ip = nautobot.device_oob_ip(device_id)
+    oob = oob_sushy_session(oob_ip, bmc_user, bmc_pass)
```

Comment:
ah I didn't see your comment before I hit merge. I'm trying to follow the terminology used in industry which is BMC

Reply:
No worries - In my eyes BMC is just an "implementation" of [OOB](https://en.wikipedia.org/wiki/Out-of-band_management), or more specifically a name of hardware used to provide OOB services.

I don't mind using either, but let's keep it consistent, for example instead of this:

```python
def do_sync(device_id: UUID, nb_url: str, nb_token: str, bmc_user: str, bmc_pass: str):
    nautobot = Nautobot(nb_url, nb_token, logger=logger)
    oob_ip = nautobot.device_oob_ip(device_id)
    oob = oob_sushy_session(oob_ip, bmc_user, bmc_pass)
```

do this:


```python
def do_sync(device_id: UUID, nb_url: str, nb_token: str, bmc_user: str, bmc_pass: str):
    nautobot = Nautobot(nb_url, nb_token, logger=logger)
    bmc_ip = nautobot.device_bmc_ip(device_id)
    bmc = bmc_sushy_session(bmc_ip, bmc_user, bmc_pass)

```